src/bp/core/repositories/users.ts

```diff
@@ -81,14 +81,19 @@ export class KnexUserRepository implements UserRepository {
     return this.database.knex.json.get(user.attributes)
   }
 
-  async updateAttributes(channel: string, user_id: string, attributes: any): Promise<void> {
+  async updateAttributes(channel: string, user_id: string, attributes: any, overwrite?: boolean): Promise<void> {
     channel = channel.toLowerCase()
 
     if (this.dataRetentionService.hasPolicy()) {
       const originalAttributes = await this.getAttributes(channel, user_id)
       await this.dataRetentionService.updateExpirationForChangedFields(channel, user_id, originalAttributes, attributes)
     }
 
+    if (!overwrite) {
```

Comment:
why do we allow overwrite ? if you want to overwrite just pass the whole attributes object in the payload ..

Reply:
because you can edit one or multiple attributes without erasing all the other ones, while in other cases, you want to handle all of the attributes (for example, in the data retention service)